src/main/java/gregtech/api/capability/impl/miner/MinerLogic.java

```diff
@@ -0,0 +1,645 @@
+                    // check every block along the x-axis
+                    if (x.get() <= startX.get() + currentRadius * 2) {
+                        BlockPos blockPos = new BlockPos(x.get(), y.get(), z.get());
+                        Block block = metaTileEntity.getWorld().getBlockState(blockPos).getBlock();
+                        if (block.blockHardness >= 0 && metaTileEntity.getWorld().getTileEntity(blockPos) == null && GTUtility.isOre(block)) {
+                            blocks.addLast(blockPos);
+                        }
+                        // move to the next x position
+                        x.incrementAndGet();
+                    } else {
+                        // reset x and move to the next z layer
+                        x.set(x.get() - currentRadius * 2);
+                        z.incrementAndGet();
+                    }
+                } else {
+                    // reset z and move to the next y layer
+                    z.set(z.get() - currentRadius * 2);
+                    y.decrementAndGet();
+                }
+            } else
+                return blocks;
+
+            // only count iterations where blocks were found
+            if (!blocks.isEmpty())
+                calculated++;
+        }
+        return blocks;
+    }
+
+    /**
+     * gets the quotient for determining the amount of blocks to mine
+     * @param base is a value used for calculation, intended to be the mean tick time of the world the miner is in
+     * @return the quotient
+     */
+    private static double getQuotient(double base) {
+        return DIVIDEND / Math.pow(base, POWER);
+    }
+
+    /**
+     * @param values to find the mean of
+     * @return the mean
+     */
+    private static long mean(long[] values) {
+        long sum = 0L;
+        for (long v : values)
+            sum += v;
+        return sum / values.length;
+    }
+
+    /**
+     *
+     * @param world the {@link World} the miner is in
+     * @return the mean tick time
+     */
+    private static double getMeanTickTime(World world) {
+        return mean(Objects.requireNonNull(world.getMinecraftServer()).tickTimeArray) * 1.0E-6D;
+    }
+
+    /**
+     * Applies a fortune hammer to block drops based on a tier value, intended for small ores
+     * @param random the random chance component for recipe ChancedOutputs
+     * @param blockState the block being mined
+     * @param drops where the drops are stored to
+     * @param fortuneLevel the level of fortune used
+     * @param player used to determine whether there is a fake player being used
+     * @param map the recipemap from which to get the drops
+     * @param tier the tier at which the operation is performed, used for calculating the chanced output boost
+     */
+    protected static void applyTieredHammerNoRandomDrops(Random random, IBlockState blockState, List<ItemStack> drops, int fortuneLevel, EntityPlayer player, RecipeMap<?> map, int tier) {
```

Comment:
The `player` being passed into this method is always `null`. Is this parameter needed?